src/lib/core/ReferenceCounted.h

```diff
@@ -62,7 +70,7 @@ class ReferenceCounted
 
         if (--mRefCount == 0)
         {
-            delete this;
+            DELETOR::Release(reinterpret_cast<SUBCLASS *>(this));
```

Comment:
static_cast should be enough if SUBCLASS is really a subclass

Reply:
it is `reinterpret_cast` in `Retain` function a few lines above